test/test_pty.rs

```diff
@@ -0,0 +1,36 @@
+use nix::pty::openpty;
+use nix::unistd::{read, write, close};
+
+#[test]
+fn test_openpty() {
+    // TODO: figure out the right termios settings to pass in here
```

Comment:
You can only use `tcgetattr` on ttys, which I don't think stdin is, which explains why it fails on mac. What I would recommend is calling `tcgetattr` on one of the pttys you create and then use that to open one with that termios. So you'd do both of these in the same test. I'd make that a separate test than the one you already have that is testing reading & writing from a master/slave pair.